### ISSUE TITLE
Reference document: Implement Federal Awards screen

### ISSUE THREAD

**Issue by MelissaBraxton** [GSA-TTS/FAC#187]
## Background

This issue covers the front end and back end implementation of the SEFA page of SF-SAC data entry. You can find this in the story map [here](https://app.mural.co/t/gsa6/m/gsa6/1647625636085/7825fa5ead63ebb1b65eefab5ea20d24a8ab7c59?wid=0-1647626680995). 

**Stories**

1. As an auditee, I can enter or edit information on my federal awards and have it populate my SEFA so that I can complete my audit.
2. As an auditee for an entity with a high volume of awards, I can bulk upload federal award data in an acceptable format to save time. 
3. As an auditee, I want an easy way to share my completed SEFA with my auditor to aid them in their audit. 

**Acceptance criteria (We'll know we're done when...)**
- [ ] The page renders as designed
- [ ] A user can download [this worksheet](https://docs.google.com/spreadsheets/d/1GEhEUPlsMFn7yFV4N2KAeCnKNom-crtI/edit#gid=459082746) ~~with their EIN number inside~~
- [ ] A user can upload a completed worksheet via drag and drop or file selection
- [ ] Display worksheet validation errors
  - What is list of errors? 
- [ ] Display file name in place of "upload worksheet" button when it validates successfully. 
- [ ] Display option to delete uploaded file with warning message
- [ ] When "Save and continue" button is clicked, data is uploaded to the server and saved in the DB
  
**Tasks**
- [ ] https://github.com/GSA-TTS/FAC/issues/221#issuecomment-1145177220
- [ ] Front end for SEFA page
- [ ] Backend for worksheet upload and download

**Definition of Done**
- [ ] Meets acceptance criteria
- [ ] Screen reader - Listen to the experience with a screen reader extension, ensure the information presented in order
- [ ] Keyboard navigation - Run through acceptance criteria with keyboard tabs, ensure it works. 
- [ ] Text scaling - Adjust viewport to 1280 pixels wide and zoom to 200%, ensure everything renders as expected. Document 400% zoom issues with USWDS if appropriate. 

## Design

https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=3159%3A20568

![image](https://user-images.githubusercontent.com/2846578/182708679-38f187a9-2a39-4853-abbc-df5372dbb8c6.png)

## Implementation Notes

- For the prototype only the Federal awards page has the modal states, but it's going to be the same throughout for implementation (Award findings, Findings text, CAP text, Additional EINs, Additional UEIs, Secondary auditors)
- It will use the [USWDS file input component](https://designsystem.digital.gov/components/file-input/ ) and the default functionality
- Please note: the top-level navigation and the page header (organization name, report id, entity type, etc.) layout will be changing in the future.

See https://github.com/GSA-TTS/FAC/issues/255 for additional discussion.

## Copy

https://docs.google.com/document/d/1rCdqfW5XSVBp3a768Z6Ql4a-g9s0EfxGKyH-4uD1SYc/edit#heading=h.tjrx982norah

**Comment by petermarks-gsa** [comment]
@melchoyce Can you please confirm:

1) [this figma link](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=3136%3A18110) is the most up-to-date design? 
2) The "With 2 options - DSF - Federal Awards" mockup at that figma link, with PDF + Excel file upload, is NOT what we're creating here?

**Comment by austinhernandez** [comment]
Theres no 2 options anymore, that's how it was originally and thats now out of scope. i deleted it, now the link you shared is inactive. the link i shared above with the screenshots is up to date

https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=3159%3A20568

**Comment by melchoyce** [comment]
Most up-to-date version of the mockups here: https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=4954%3A23338&t=TixnjgpGZuY87dAV-4

**Comment by pkigongo** [comment]
Blocked by #309
